wallet/src/de/schildbach/wallet/ui/MainActivityViewModel.kt

```diff
@@ -40,7 +40,7 @@ class MainActivityViewModel(application: Application) : AndroidViewModel(applica
     val canAffordIdentityCreationLiveData = CanAffordIdentityCreationLiveData(walletApplication)
 
     val hasIdentity: Boolean
-        get() = blockchainIdentity?.creationState == BlockchainIdentityData.CreationState.DONE
+        get() = blockchainIdentity?.creationComplete!!
```

Comment:
not sure why we need !! here, but the compiler complains about it.

Reply:
This i due to the safe call operator `?.` after `blockchainIdentity`. It would return `null` if `blockchainIdentity` is `null`. Your fix should be interpreted as `(blockchainIdentity?.creationComplete)!!` so that the not-null assertion operator `!!` refers to the whole result, not only to the `creationComplete` part.
Safer solution, that will not cause NPE when getting the `hasIdentity` field for `blockchainIdentity` being `null` would be:
```
val hasIdentity: Boolean
    get() = blockchainIdentity?.creationComplete ?: false
```

Reply:
Thanks for the explanation.